analysis/filter/StructureModifier/src/main/kotlin/de/maibornwolff/codecharta/filter/structuremodifier/ParserDialog.kt

```diff
@@ -15,40 +16,58 @@ class ParserDialog {
             val inputFolderName =
                     KInquirer.promptInput(
                             message = "What is the cc.json file that has to be modified?",
-                            hint = Paths.get("").toAbsolutePath().toString() + File.separator + "yourInput.cc.json"
-                                         )
+                            hint = Paths.get("").toAbsolutePath().toString() + File.separator + "yourInput.cc.json")
 
-            val outputFileName: String = KInquirer.promptInput(
-                    message = "What is the name of the output file?"
+            val selectedAction: String = KInquirer.promptList(
+                    message = "Which action do you want to perform?",
+                    choices = listOf(
+                            StructureModifierAction.PRINT_STRUCTURE.descripton,
+                            StructureModifierAction.EXTRACT_PATH.descripton,
+                            StructureModifierAction.MOVE_NODES.descripton,
+                            StructureModifierAction.REMOVE_NODES.descripton
+                    )
             )
 
-            val setRoot: String =
-                    KInquirer.promptInput(message = "What path within project to be extracted? (Optional)", default = "")
+            return when (selectedAction) {
+                StructureModifierAction.PRINT_STRUCTURE.descripton -> collectPrintArguments(inputFolderName)
```

Comment:
all four new methods take the inputFolderName as input but do not use it, only return it as part of the list. I think this should be solved differently